src/framework/theme/components/toggle/toggle.component.ts

```diff
@@ -0,0 +1,271 @@
+ * toggle-danger-focus-border-color:
+ * toggle-danger-hover-background-color:
+ * toggle-danger-hover-border-color:
+ * toggle-danger-active-background-color:
+ * toggle-danger-active-border-color:
+ */
+@Component({
+  selector: 'nb-toggle',
+  animations: [
+    trigger('onOff', [
+      state('ltrOn', ltrState, defaultState),
+      state('rtlOn', rtlState, defaultState),
+      transition(':enter', [animate(0)]),
+      transition('* <=> *', [animate('0.25s')]),
+    ]),
+  ],
+  template: `
+    <label class="toggle-label">
+      <input type="checkbox" class="native-input visually-hidden"
+             [disabled]="disabled"
+             [checked]="checked"
+             (change)="updateValue($event)"
+             (blur)="setTouched()"
+      >
+      <div class="toggle" [class.checked]="checked">
+        <span [@onOff]="checkState()" class="toggle-switcher">
+          <nb-icon *ngIf="checked" icon="checkmark-bold-outline" pack="nebular-essentials"></nb-icon>
+        </span>
+      </div>
+    </label>
+  `,
+  styleUrls: [ `./toggle.component.scss` ],
+  providers: [{
+    provide: NG_VALUE_ACCESSOR,
+    useExisting: forwardRef(() => NbToggleComponent),
+    multi: true,
+  }],
+})
+export class NbToggleComponent implements OnInit, ControlValueAccessor {
+
+  onChange: any = () => { };
+  onTouched: any = () => { };
+
+  private direction: NbLayoutDirection;
+
+  /**
+   * Toggle checked
+   * @type {boolean}
+   * @private
+   */
+  @Input()
+  get checked(): boolean {
+    return this._checked;
+  }
+  set checked(value: boolean) {
+    this._checked = value;
+  }
+  private _checked: boolean = false;
+
+  /**
+   * Controls input disabled state
+   */
+  @Input()
+  get disabled(): boolean {
+    return this._disabled;
+  }
+  set disabled(value: boolean) {
+    this._disabled = convertToBoolProperty(value);
+  }
+  private _disabled: boolean = false;
+
+  /**
+   * Toggle status.
+   * Possible values are: `primary` (default), `success`, `warning`, `danger`, `info`
+   */
+  @Input()
+  status: '' | NbComponentStatus = '';
+
+  /**
+   * Output when checked state is changed by a user
+   * @type EventEmitter<boolean>
+   */
+  @Output() checkedChange = new EventEmitter<boolean>();
+
+  @HostBinding('class.status-primary')
+  get primary() {
+    return this.status === 'primary';
+  }
+
+  @HostBinding('class.status-success')
+  get success() {
+    return this.status === 'success';
+  }
+
+  @HostBinding('class.status-warning')
+  get warning() {
+    return this.status === 'warning';
+  }
+
+  @HostBinding('class.status-danger')
+  get danger() {
+    return this.status === 'danger';
+  }
+
+  @HostBinding('class.status-info')
+  get info() {
+    return this.status === 'info';
+  }
+
+  constructor(
+    private changeDetector: ChangeDetectorRef,
+    private layoutDirection: NbLayoutDirectionService,
+  ) {}
+
+  ngOnInit() {
+    this.layoutDirection.onDirectionChange().subscribe(() => {
```

Comment:
Please add `takeUntil(this.destroy$)` here to unsubscribe when component destroyed.